File: code/Java/interpreter/expr/NegIntExpr.java
package interpreter.expr;

public class NegIntExpr extends IntExpr {

    private IntExpr expr;

    public NegIntExpr(int line, IntExpr expr) {
        super(line);
        this.expr = expr;
    }

    public int expr() {
        return -expr.expr();
    }

}

File: code/C++/interpreter/expr/NegIntExpr.cpp
#include "NegIntExpr.h"

NegIntExpr::NegIntExpr(int line, IntExpr* expr)
	: IntExpr(line), m_expr(expr) {
}

NegIntExpr::~NegIntExpr() {
	delete m_expr;
}

int NegIntExpr::expr() {
	return -m_expr->expr();
}
